src/CommunityToolkit.Maui.SourceGenerators.Internal/Models/Records.cs

```diff
@@ -0,0 +1,22 @@
+﻿using CommunityToolkit.Maui.SourceGenerators.Internal.Helpers;
+using Microsoft.CodeAnalysis;
+
+namespace CommunityToolkit.Maui.SourceGenerators.Internal.Models;
+
+record BindablePropertyModel
+{
+	public string PropertyName { get; set; } = string.Empty;
```

Comment:
### 1. Add `record XMLTagModel`

I suggest we create a new `record XMLTagModel` and add it to `BindablePropertyModel` as a nullable object. If `BindablePropertyModel.XMLTags` is `null`, then the source generator won't add any XML comments. If it is instantiated, then the source generator will generate the XML documentation for the non-null strings in `XMLTagModel`. 

The nullable string ensures we don't accidentally generate a blank XML comment for `Summary`, for example.

```cs
record XMLTagModel(string? Summary, string? Remarks, string? Returns, string? Param, string? Paramref, string? ExceptionDescription);

record BindablePropertyModel
{
        public XMLTagModel? XMLTags { get; set; }
	public string PropertyName { get; set; } = string.Empty;
// ...
}
```

### 2. Add XML Tag `string?`s to `BindablePropertyModel`

If using an object in the attribute doesn't look/feel right, we can add each XML Tag option to `BindablePropertyModel` as a nullable string.

The nullable string ensures we don't accidentally generate a blank XML comment for `Summary`, for example.

```cs
record BindablePropertyModel
{
	public string? Summary { get; set; }, 
	public string? Remarks { get; set; }
	public string? Returns { get; set; }
	public string? Param { get; set; }
	public string? Paramref { get; set; }
	public string? ExceptionDescription { get; set; }
	public string PropertyName { get; set; } = string.Empty;
// ...
}
```

Then the source generate can add the XML.

If our BindableProperty Source Generator doesn't support XML tags, then we can't use it in the Toolkit because we have so many BindableProperties with unique XML tags.

Reply:
For 1: 
The usage of a custom type will not work, because it will not be a valid type for attribute, as you can see [here](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/attributes#2224-attribute-parameter-types).

For 2:

That will cause a lot of noise on the attribute, making it very big. Don't think the best solution will be using C#, I know that SG can consume xml and json files. The usage of xml files for docs is already used on Essentials and Maui as well, maybe we can mimic that with SG?